PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+Secondly, verifying the correctness and self-consistency of data is significantly less complex than verifying the correctness of a program's behaviour, which is fundamentally undecidable. 
+Verifying that a set of declarations is correct, however, _is_ decidable, even if sometimes computationally expensive.
+As a general rule, this implies that the data language used to defined the Desired State should have no control flow structure, and consist exclusively of referentially transparent expressions without side effects. 
+In other words, a human-readable data-serialization format.
+
+It is also preferrable to use a widely supported language to define the Desired State for the sake of interoperability. For example, YAML, XML, JSON and TOML all have broad support and well defined specifications, although there are many other suitable candidates.
+
+#### Why is human readability required?
+
+Operationally, it has been proven time and time again that the canonical Desired State of a system should be human-readable and writable.
+
+The Desired State encodes the _intent_ of a system. For example, where internet traffic should be routed. The collection of decisions about a system captured in its Desired State are of particular interest to its human operators. They must be able to directly describe their intent about the state of a system; and similarly, must be able to decipher the intended state of a system from its configuration.
+
+This principles not only capture a requirement about the format of the configuration, but also the qualitative readability of configuration _in practice_. For example, a simple YAML or XML file can be easily read, understood and modified by human operators, but experience has taught us that these formats also allow the creation of complex self-referential documents beyond the ability of most humans to interpret. Such complex document would violate this principles, even though the formats are, on the surface, human-readable.
+
+This relative readability also implies that the users are relevant when evaluating whether a system follows this principle. For example, a Desired State defined with a rich grammar of S-Expressions would suite a team of developers with a background in functional programming, but would violate this principle if the majority of the team found the format incomprehensible.
+
+Having a human-readable Desired State does not in any way preclude the use of rich tooling or graphical interfaces that facilitate Desired State generation and interpretation. It only require that the canonical source of truth be human readable and writable plain text.
+
+#### How much of a system must be declared?
+
+Ideally, all of it; and the entire system can be recreated exclusively from its Desired State.
+
+That said, the definition of a "computer systems" can be quite broad and strays into human processes and systems as well. 
```

Comment:
```suggestion
The definition of _system_ can be quite broad, and may incorporate human as well as programmatic processes. Although a vision in which the GitOps practices are applied generally to all processes, human or otherwise, that participate in any one system, a more pragmatic approach is preferable, to avoid the risk of attempting to "boil the ocean".
```